contracts/src/Teleporter/TeleporterMessenger.sol

```diff
@@ -731,19 +749,35 @@ contract TeleporterMessenger is ITeleporterMessenger, ReentrancyGuards {
             return;
         }
 
-        // Call the destination address of the message with the formatted call data.
-        // Only provide the required gas limit to the sub-call so that the end application
-        // cannot consume an arbitrary amount of gas.
-        // solhint-disable-next-line avoid-low-level-calls
-        (bool success, ) = message.destinationAddress.call{
-            gas: message.requiredGasLimit
-        }(
-            abi.encodeCall(
-                ITeleporterReceiver.receiveTeleporterMessage,
-                (originChainID, message.senderAddress, message.message)
-            )
+        // Encode the payload by ABI encoding a call to the {receiveTeleporterMessage} function
+        // defined by the {ITeleporterReceiver} interface.
+        bytes memory payload = abi.encodeCall(
+            ITeleporterReceiver.receiveTeleporterMessage,
+            (originChainID, message.senderAddress, message.message)
         );
 
+        // Call the destination address of the message with the formatted call data. Only provide the required
+        // gas limit to the sub-call so that the end application cannot consume an arbitrary amount of gas.
+        //
+        // Use assembly to make the low-level call to avoid unnecessary memory expansion of the return data
+        // due to an issue with the version of Solidity used. This prevents possible "return bomb" vectors
+        // where the external contract could for the caller to use an arbitrary amount of gas.
```

Comment:
Same awkward wording here.

Reply:
Updated 👍 Let me know if you think it's clearer now or can think of a better wording.